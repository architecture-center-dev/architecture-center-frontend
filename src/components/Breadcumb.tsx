import React, { SyntheticEvent } from 'react';
import Breadcrumbs from '@material-ui/core/Breadcrumbs';
import Link from '@material-ui/core/Link';

function handleClick(event : SyntheticEvent) {
  event.preventDefault();
  console.info('You clicked a breadcrumb.');
}

const BreadCumbs = () => {
  return (
    <Breadcrumbs aria-label="breadcrumb">
      <Link color="inherit" href="/" onClick={handleClick}>
        Home Page
      </Link>
      <Link color="inherit" href="/getting-started/installation/" onClick={handleClick}>
        Solution
      </Link>
    </Breadcrumbs>
  );
}

export default BreadCumbs;