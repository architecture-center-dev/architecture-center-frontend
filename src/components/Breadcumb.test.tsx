import * as React from "react";
import Breadcumb from "./Breadcumb";
import { render, fireEvent } from '@testing-library/react'

describe("<Breadcumb />", () => {
  it("Component should render correctely", async () => {

    const { getByText } = render(<Breadcumb />);
    const breadcumb = getByText('Solution');

    fireEvent.click(breadcumb);

    expect(breadcumb).toBeInTheDocument;
    expect(breadcumb).toMatchSnapshot();
  });
});
