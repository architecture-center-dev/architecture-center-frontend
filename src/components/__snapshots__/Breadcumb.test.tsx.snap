// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcumb /> Component should render correctely 1`] = `
<a
  class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
  href="/getting-started/installation/"
>
  Solution
</a>
`;
